### ISSUE TITLE
logpath for logoutput

### ISSUE THREAD

**Issue by unclesamwk** [restic/restic#1455]
Hey,

an very nice feature was an logpath and some logging features.

```restic -r xyz backup ~/home --log /tmp/job1.log```

greetings
Sam

**Comment by fd0** [comment]
Hey, thanks for the suggestion. We don't have a clear concept for logging yet.

**Comment by whereisaaron** [comment]
The current output is interactive with line breaks and other stuff that doesn't suit logging or scripting. It would be nice to have some basic logging and a `--batch` switch or similar (could be automatic if `stdin` is not a tty) to output the log messages and turned off the interactive stuff. Same info, just without the line breaks :-)

**Comment by CristianCantoro** [comment]
I agree with @whereisaaron, here's a script that I set up to work around that: [`backup_restic.sh`][1].
(note that I have defined `$RESTIC_BACKUP_TARGET` and `$RESTIC_EXCLUDES_FILE`, in addition to`$RESTIC_REPOSITORY` and `$RESTIC_PASSWORD_FILE`

Here's a sample output [from pastebin][2],

It is not perfect because the parts of the output that are changing don't get printed, as they are produced.
The `awk` script should be able to handle that, but it is not working as I expected.

Also, I can not understand why it does not print `password is correct`, that get printed if I launch the command from the CLI without redirections.

[1]: https://gist.github.com/CristianCantoro/8cc103ef90c6a257f5eed365c7832da2#file-backup_restic-sh
[2]: https://pastebin.com/raw/1y2KarRy

**Comment by CristianCantoro** [comment]
An addition or an alternative to what @whereisaaron is proposing is having the possibility to print progress information when a signal is received. This is how, for example how [`tar`][1] shows progress (see the `--totals` flag).

[1]: https://www.gnu.org/software/tar/manual/html_section/tar_25.html

**Comment by whereisaaron** [comment]
@fd0 the current output is really only suited to running backups by hand, interactively. But who does that? Noone but the devs :-)  Everyone schedules their backups and just wants a record of what happened, or to stream log output into log monitoring tools to get an alert when something bad is in the log.

If you don't have a concept yet, then I'd suggest jump to the current status quo:
1. Leave the current output if stdout is a tty or if `--interactive` flag
2. Log to stdout if no tty or if `--noninteractive` flag
3. Use structured logging in JSON with a simple structured logger like [Uber's 'zap'](https://github.com/uber-go/zap)
4. Don't have log file arguments or try to manage own logs, [just stream them to stdout](https://12factor.net/logs)

**Comment by frittentheke** [comment]
@whereisaaron nice suggestions. If running "noninteractive", so i.e. as systemd.timer job, at certain intervals the progress should be reported (as new line, not like in interactive) though. Otherwise it's hard to debug a very long running backup and to see how it progresses. Kinda like what @CristianCantoro does with his script.

**Comment by whereisaaron** [comment]
@frittentheke I would suggest that is not the mechanism, that's the log level. If you enable 'debug' level logging, interactive or not interactive, then I'd expect every file and/or uploaded block would be logged. 

If you are not debugging, and the job is non-interactive, then if a job isn't starting, stopping, timing out, or experiencing an error, then there is nothing interesting to say or worthy of logging.

None of this precludes things like a `--totals` signal mechanism. By all means have them too. If a noninteractive job seems stuck, it is great to have a mechanism to poke them for a status dump.

**Comment by fd0** [comment]
I'd like to defer the discussion until #1494 is done. I'm currently reworking the archiver code, which includes the reporting system for errors, status and totals. Let's come back to this issue after it is done. Thanks!

**Comment by whereisaaron** [comment]
@CristianCantoro the new 0.9.0 release  has totals, there is no `--totals` option or signal mechanism, but with `--verbose` you get totals at the end, e.g.

```
Files:          14 new,    16 changed, 43200 unmodified
Dirs:            0 new,     2 changed,     0 unmodified
Data Blobs:    290 new
Tree Blobs:      3 new
Added:      261.509 MiB

processed 43230 files, 51.587 GiB in 11:30
```

**Comment by HaleTom** [comment]
Given #1494 is now merged and I would like to enable logging for #2270, would it perhaps be time to resume discusison?

**Comment by xxxmicrobexxx** [comment]
> 
> 
> Given #1494 is now merged and I would like to enable logging for #2270, would it perhaps be time to resume discussion?

Yes please.  Logging is still a fairly arcane process.

**Comment by PrplHaz4** [comment]
> > Given #1494 is now merged and I would like to enable logging for #2270, would it perhaps be time to resume discussion?
> 
> Yes please. Logging is still a fairly arcane process.

It appears work was started in PR #2436, but that wasn’t merged.
